src/protonpack/java/com/codepoetics/protonpack/AggregatingSpliterator.java

```diff
@@ -15,8 +17,8 @@
     private List<I> currentSlide = new ArrayList<>();
 
     AggregatingSpliterator(Spliterator<I> source, BiPredicate<List<I>, I> predicate) {
-        this.source = source;
-        this.condition = predicate;
+        this.source = requireNonNull(source);
+        this.condition = requireNonNull(predicate);
```

Comment:
I think I will in this case. I'm not convinced of the value of null checks. Far better not to go passing nulls around in the first place. If you pass null as a parameter into something, my view is that you deserve everything you get. A standardised @NotNull annotation, compiler-checked, would be worth having; but my preference would be for non-null to be standard, and nullibility to be clearly marked as an exception.